psychopy/hardware/bbtk/__init__.py

```diff
@@ -242,6 +241,101 @@ def parseEventsLine(line, lastState=None):
         logging.flush()  # we aren't in a time-critical period
         return events
 
+    def setResponse(self, sensor=None, outputPin = None, testDuration = None,
+                    responseTime=None, nTrials=None, setSmoothing = True,
+                    responseDuration = None):
+        """
+        Sets Digi Stim Capture and Response (DSCAR) for BBTK.
+
+        :param sensor: Takes string for single sensor, and tuple or list of strings for multiple sensors
+        :param outputPin: Takes string for single output, and tuple or list of strings for multiple outputs
+        :param testDuration: The duration of the testing session in seconds
+        :param responseTime: Time in seconds from stimulus capture that robotic actuator should respond
+        :param nTrials: Number of trials for testing session
+        :param setSmoothing: For use with CRT monitors - Defaults to False for common LCD screens
+        :param responseDuration: Time in seconds that robotic actuator should stay activated for each response
+        :return:
+        """
+        # Create sensor and outputPin dicts
+        sensorDict = dict(zip(
+            ['keypad4', 'keypad3', 'keypad2', 'keypad1', 'opto4',
+             'opto3', 'opto2', 'opto1', 'ttlin2', 'ttlin1', 'mic2','mic1'],
+            [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]))
+        outputDict = dict(
+            zip(['actclose4', 'actclose3', 'actclose2', 'actclose1', 'ttlout2', 'ttlout1', 'sounder2', 'sounder1'],
+                [0, 1, 2, 3, 4, 5, 6, 7]))
+        # Check parameters
+        if sensor is None:
+            print("Setting BBTK pattern matching to 'INDI' - respond to any trigger")
```

Comment:
In general I'd suggest using the logging module for this rather than print() so that the user can control the level of messages. Maybe logging.info() for this one?

Reply:
When I say the logging module I mean psychopy.logging not the built-in logging module

Reply:
Cheers Jon, no problem, all done.